embedded-graphics/src/style/primitive_style.rs

```diff
@@ -70,6 +70,23 @@ where
     pub(crate) fn stroke_width_i32(&self) -> i32 {
         i32::try_from(self.stroke_width).unwrap_or(i32::max_value())
     }
+
+    /// Returns the effective stroke width.
+    ///
+    /// The effective stroke width takes the stroke color into account. If `stroke_color` is `None`
+    /// the effective stroke width is `0` independent of `stroke_width`.
+    pub(crate) fn effective_stroke_width(&self) -> u32 {
+        if self.stroke_color.is_some() {
+            self.stroke_width
+        } else {
+            0
+        }
```

Comment:
This could be:

```suggestion
        self.stroke_color.map(|_| self.stroke_width).unwrap_or(0)
```

Reply:
TBH I think the original version is more readable.

Reply:
Fair! Not worth bikeshedding over :)